### PR TITLE
Maximum profit

### PR DESCRIPTION
**Pull request by hridhi**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes 

### Description

I have written the code in python and java for maximum profit problem.[Dynamic programming]

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
https://github.com/TesseractCoding/NeoAlgo/issues/3778

### CODE REVIEW

**Comment by hridhi** [comment]
This is a part of GSSoC'21.Please review my PR. @HarshCasper @siddharth25pandey

File: Python/dp/Maximum_Profit.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	result = profit[n-1]
+
+	return result
+
+# Driver function
+for _ in range(int(input())):
+    price = []
+    length=int(input())
+    price=list(map(int,input().split()))#90,80,70,60,50
+    print ("Maximum profit is", maxProfit(price,length))
```
Comment (HarshCasper) [inline Python/dp/Maximum_Profit.py]: Add a new line at the EOF

File: Python/dp/Maximum_Profit.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+Method :
+Dynamic programming 
+
+Intuition : 
+Storing the maximum possible profit of every subarray in a table named profit 
+Finally returning the profit[n-1]
+
+
+Time complexity : O(N)
+Space Complexity : O(N)
```
Comment (HarshCasper) [inline Python/dp/Maximum_Profit.py]: Shift Time & Space Complexity at the end of the Code

File: Python/dp/Maximum_Profit.py (hunk 1 above)
Comment (HarshCasper) [inline Python/dp/Maximum_Profit.py]: Give sample I/O as well

File: Python/dp/Maximum_Profit.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+		if price[i] < min_price:
+			min_price = price[i]
+
+		# Maximum profit is maximum of:
+		# a) previous maximum,
+		# i.e., profit[i-1]
+		# b) (Buy, Sell) at
+		# (min_price, A[i]) and add
+		# profit of other trans.
+		# stored in profit[i]
```
Comment (HarshCasper) [inline Python/dp/Maximum_Profit.py]: Use multi-line strings

File: Python/dp/Maximum_Profit.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+		# i.e., profit[i+1]
+		# b) profit by buying at
+		# price[i] and selling at
+		# max_price
+		profit[i] = max(profit[i+1], max_price - price[i])
+
+	# Get the maximum profit
+	# with two transactions allowed
+	# After this loop, profit[n-1]
+	# contains the result
```
Comment (HarshCasper) [inline Python/dp/Maximum_Profit.py]: Use multi-line strings

File: Python/dp/Maximum_Profit.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+		if price[i] > max_price:
+			max_price = price[i]
+
+		# we can get profit[i] by
+		# taking maximum of:
+		# a) previous maximum,
+		# i.e., profit[i+1]
+		# b) profit by buying at
+		# price[i] and selling at
+		# max_price
```
Comment (HarshCasper) [inline Python/dp/Maximum_Profit.py]: Use multi-line strings

File: Python/dp/Maximum_Profit.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+	# Create profit array and initialize it as 0
+	profit = [0]*n
+
+	# Get the maximum profit
+	# with only one transaction
+	# allowed. After this loop,
+	# profit[i] contains maximum
+	# profit from price[i..n-1]
+	# using at most one trans.
```
Comment (HarshCasper) [inline Python/dp/Maximum_Profit.py]: Use multi-line strings

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: Python/dp/Maximum_Profit.py (hunk 1 above)
Comment (hridhi) [reply Python/dp/Maximum_Profit.py:87]: didn't get you

Comment (hridhi) [reply Python/dp/Maximum_Profit.py:87]: it automatically prints in next line for different test cases

**Comment by hridhi** [comment]
Made changes please check @HarshCasper

**Review by HarshCasper — APPROVED** [review]

**Comment by HarshCasper** [comment]
@all-contributors bot please add @hridhi for code

**Pull request merged**